### ISSUE TITLE
Questions about replicating m2m model

### ISSUE THREAD

**Issue by yumoxu** [csebuetnlp/CrossSum#4]
Hi there, happy new year and thanks for releasing the code for your nice work! I have some questions about the training configs and I hope you could clarify them for me. 

Specifically, I would like to replicate the released m2m model and I am using the provided `trainer.sh`. As I use two A100s, I set `PER_DEVICE_TRAIN_BATCH_SIZE=32` so as to keep the effective batch size to 256. I keep the rest configs intact. 

Then I do inference on the `Chinese Simplified-English` test set with the checkpoint at the 25K step. It gets **21.65** on ROUGE-L, while the released m2m model gets **26.75**. 

After inspecting the model outputs, I found that the model of my replication sometimes generates summaries of non-target languages. For example, for the `Chinese Simplified-English` test set, around 10% of the generated summaries are Chinese, while the released model is able to generate only English summaries. This may explain the above performance gap. 

Another observation is that the above-mentioned problem is more severe with the checkpoint at the 20K step, so I wonder if this is due to underfitting, and it may vanish with more training steps (e.g., 30K). I have not validated this assumption yet as I would like to adopt your original training configs if possible. 

I would appreciate it if you could shed some light on how to correctly replicate your m2m model. Are there any particular training configs that I should adopt? It would also help if you could share which checkpoint (training steps) the released m2m model is. 

Many thanks!

**Comment by Tahmid04** [comment]
Hi,

Thanks for using CrossSum for your work! 

The released model was trained with the same code in this repository without any changes (i.e., the released checkpoint was trained for 25k steps). Even with the changes in `PER_DEVICE_TRAIN_BATCH_SIZE`, the results are not supposed to drop by this margin if the other hyperparameters were unchanged during training and inference. 

We tried training with A100 GPUs having 40 GB VRAM, but failed to load more than 8 samples per GPU in an instant. Did you make any other changes in the config like reducing the max input/output length, changing the optimizer, or FP16 training? Sorry for asking, you already mentioned that no other changes were made; but asking just to be absolutely sure.

**Comment by yumoxu** [comment]
Thanks for your clarification. I have double-checked my `trainer.sh`, and can confirm `PER_DEVICE_TRAIN_BATCH_SIZE` is the only parameter I have changed. 

Quite interesting to know that A100 80GB can fit 32 samples while A100 40GB takes 8 -- I did not alter the max input/output length/optimizer/FP16. 

The same inference script produces reasonable results on the released model, so the inference configs should be okay.

If my understanding is right, there are no other training configs that rely on the number of GPUs, e.g.,  `warmup_steps`/`gradient_accumulation_steps`/`per_lang_batch_size`. Could you confirm this?

**Comment by Tahmid04** [comment]
We had a guess that you were using 80 GB GPUs. Unfortunately, it is not possible to reproduce your setting with our resources. The only thing we can suggest is using  `PER_DEVICE_TRAIN_BATCH_SIZE=8` and `gradient_accumulation_steps=16`. The other hyperparameters in the `trainer.sh` script can remain as is.

**Comment by yumoxu** [comment]
Great suggestion, I will give it a try. Thanks again!

**Comment by yumoxu** [comment]
@Tahmid04 It seems `pytorch==1.7.1` recommended in README is incompatible with `cudatoolkit=11.1`. Installing these two packages with `conda` results in the installation of `cpuonly` version of `pytorch==1.7.1`, and this leads to crash in distributed training with GPUs. 

My previous results are obtained with `pytorch==1.8.1`. I am wondering if this plays a role in the performance difference, and I am trying to reproduce your environment. May I ask you which `pytorch` version did you use for A100s with `cudatoolkit=11.1`?

**Comment by Tahmid04** [comment]
We did not face such incompatibility issues during installation. We are currently using `pytorch==1.7.1` with `cudatoolkit=11.0` (not `11.1`) and distributed training is running fine on our setup. Could you please try setting up with `cudatoolkit=11.0`?

**Comment by Tahmid04** [comment]
Since there has not been any activity in over a month, I am closing this issue as completed. Feel free to reopen it for further queries, thanks.
